### PR TITLE
[DUOS-809][risk=no] Fix DAR header display

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadinstitute.atlassian.net/browse/DUOS-809

## Changes
Get the datasets directly to populate the full header of the DAR.

### New:
Screen Shot 2020-11-02 at 1 31 46 PM

### Old:
Screen Shot 2020-11-02 at 1 33 44 PM

----

Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] PR is labeled with a Jira ticket number and includes a link to the ticket
- [ ] PR is labeled with a security risk modifier [no, low, medium, high] 
- [ ] PR describes scope of changes

In all cases:

- [ ] Get a minimum of one thumbs worth of review, preferably 2 if enough team members are available
- [ ] Get PO sign-off for all non-trivial UI or workflow changes
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/DataAccessRequestHeader.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  return (
+    div({style: style}, [
+      div({}, [
+        span([projectTitle, pipe]),
+        span({}, [darCode]),
+        div({}, [
+          span([aliases, pipe]),
+          span([names, pipe]),
+          span({}, [consentName]),
```
Comment (lindojong) [inline src/components/DataAccessRequestHeader.js]: nit: a little inconsistent here with the empty style brackets

File: src/components/DataAccessRequestHeader.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    // dar.datasets.value holds the ID of the dataset
+    const datasetIds = map('value')(datasets);
+
+    async function fetchDatasets() {
+      const response = await Promise.all(
+        map((id) => {
+          return DataSet.getDataSetsByDatasetId(id);
+        })(datasetIds),
+      );
+      setFullDatasets(response);
```
Comment (lindojong) [inline src/components/DataAccessRequestHeader.js]: Not necessary, but I'm wondering if we would ever want to sort the datasets to ensure they appear in a consistent pattern for the header. I'm not sure how many datasets are typically named on a DAR for that to be useful though.

**Review by lindojong — APPROVED** [review]
LGTM!

File: src/components/DataAccessRequestHeader.js (hunk 2 above)
Comment (rushtong) [reply src/components/DataAccessRequestHeader.js]: I really like this idea. Gonna give it a shot

File: src/components/DataAccessRequestHeader.js (hunk 1 above)
Comment (rushtong) [reply src/components/DataAccessRequestHeader.js]: Done

File: src/components/DataAccessRequestHeader.js (hunk 2 above)
Comment (rushtong) [reply src/components/DataAccessRequestHeader.js]: Done

**Pull request merged**